app/assets/javascripts/discourse/app/components/sidebar/filter.gjs

```diff
@@ -0,0 +1,78 @@
+import Component from "@glimmer/component";
+import { tracked } from "@glimmer/tracking";
+import { Input } from "@ember/component";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import willDestroy from "@ember/render-modifiers/modifiers/will-destroy";
+import { inject as service } from "@ember/service";
+import DButton from "discourse/components/d-button";
+import dIcon from "discourse-common/helpers/d-icon";
+import i18n from "discourse-common/helpers/i18n";
+import { bind } from "discourse-common/utils/decorators";
+
+export default class Filter extends Component {
+  @service sidebarState;
+  @tracked filteredOutSections = this.sidebarState.filteredOutSections;
```

Comment:
I don't _think_ this is necessary... since `get displayNoResults()` is a getter, it should recalculate its value based on the value of `this.sidebarState.filteredOutSections` if you just do this directly:

```javascript
get displayNoResults() {
  return (
    this.sidebarState.filteredOutSections.length ===
    this.sidebarState.currentPanel.sections.length
  );
}
```